src/widgets/view/view.rs

```diff
@@ -157,11 +162,54 @@ impl View {
         }
     }
 
+    fn paste(&mut self) {
+        let clipboard = self.clipboard.lock().unwrap();
+        match *clipboard {
+            Some(ref content) => self.client.paste(content),
+            None => {}
+        };
+    }
+
+
+    fn cut(&mut self) {
+        let arc = self.clipboard.clone();
+        let future = self.client.cut()
+                      .and_then(move |x| { let mut clipboard = arc.lock().unwrap();
```

Comment:
I've been warned against locking a mutex inside a future, because they block the entire event loop. I think we should use something like https://github.com/asomers/futures-locks instead.

Reply:
On HEAD this is not there anymore. It goes through the same poll-loop in editor.rs as the "new_view" request